### ISSUE TITLE
Quickbooks::IntuitRequestException Validation filter broken?

### ISSUE THREAD

**Issue by Kvothe-28** [ruckus/quickbooks-ruby#555]
Hi there, I'm trying to update a quickbooks customer (to be more specific I'm tryig to create a new customer by defining customer as `customer = Quickbooks::Model::Customer.new` and then updating the Nill vallues with my own values) via values from my app's ActiveRecord model. But for some reason it raises exception errors.
At first it was raising exception error for attributes without an e-mail address;

![Screenshot_11](https://user-images.githubusercontent.com/84680882/129152404-be5acbff-8082-41ee-b9bd-db161ff36ff9.png)

Which was understandable at some point.
But then I tried to update it with an object that had an e-mail attribute, it raised this exception;

![Screenshot_10](https://user-images.githubusercontent.com/84680882/129153529-afc24bd1-5c5a-4ece-bbc6-3cf468eb18da.png)
(addresses don't contain invalid characters they are just regular letters)

When I try with different companies I still get the same error, even though e-mail address length was different error kept saying supplied length was "37 characters" which it was not. 
I can probably try to bypass this by "Query Building" but it would be my first time using it and I'd like to use it as a last resort. I'm very new in Ruby on Rails. So can anybody help me out with this? Thank you for your time.

This is how my code looks btw (running under a service). I'm trying for one object for now, after I bypass these exceptions, I'm thinking about an each do loop to iterate all companies and save them as quickbooks customers. If you think there's a better, more reliable, optimal way to do this task, please feel free to share.
![Screenshot_12](https://user-images.githubusercontent.com/84680882/129159319-053e2dc9-384a-4cc9-a62a-d9a28b062114.png)

**Comment by ruckus** [comment]
Interesting. Try enabling logging and/or debugging to see the actual request XML being generated. It could also be helpful to point your client towards an HTTP Proxy.

See: https://github.com/ruckus/quickbooks-ruby#logging

But also, the multiple assignments to `customer.billing_address` wont have much affect, the last assignment will win.

See here on how to generate a whole address with its child properties and then assign that singular address:

https://github.com/ruckus/quickbooks-ruby#physical-addresses

**Comment by Kvothe-28** [comment]
UPDATE ON THE ISSUE
@ruckus  Thank you, logging worked! Before trying your way, I tried a different approach to save customer website address exception by parsing the url and adding "http://" to the beginning.

![Screenshot_13](https://user-images.githubusercontent.com/84680882/129317216-e1c59a28-4c52-45b9-a549-5c779cfb8061.png)

I also changed format to save address like you suggested.

![Screenshot_14](https://user-images.githubusercontent.com/84680882/129317281-99ad29a0-c71f-48f3-873a-1f77e09975ef.png)

But I got this error;

![Screenshot_15](https://user-images.githubusercontent.com/84680882/129317460-30315819-6ff0-46dd-b4f3-df72cdb46b95.png)

Which is a subject for another topic. You can close this issue but can you first have a look at this error and tell me a bit more about it because I don't understand what's this is about. Thank you again.

**Comment by Kvothe-28** [comment]
UPDATE ON THE ISSUE
@ruckus  Logging didn't worked, I got "Malformed Website Address Format" error. I tried something myself but it seems it didn't worked either. I tried to validate the url by parsing then adding "http://" to the beginning.

![Screenshot_13](https://user-images.githubusercontent.com/84680882/129319279-c0ef9fc4-4271-4592-a128-d71ceb41fe1a.png)


 But it raised this error;

![Screenshot_15](https://user-images.githubusercontent.com/84680882/129319231-a53c5c5f-c33d-480e-ad65-a9fad703b39f.png)

Logging in didn't do much difference.
